File: risk-calculator-backend/risk-calculator-backend/src/main/java/com/nagarro/riskcalculatorbackend/models/CompanyDimension.java
package com.nagarro.riskcalculatorbackend.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Entity class for company dimensions
 * 
 * @author parasgautam
 *
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "company_dimensions")
public class CompanyDimension {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	@Column(name = "company_name")
	private String companyName;

	@OneToMany(cascade = CascadeType.ALL)
	@JoinColumn(name = "company_id") 
	private List<Dimension> dimensions;

}

File: risk-calculator-frontend/risk-calculator-frontend/src/app/models/CompanyDimension.ts
import { Dimension } from "./Dimension";



export class CompanyDimension {
    companyName!:string;
    dimensions!: Dimension[];

    constructor(companyName: string) {
        this.companyName = companyName;
        this.dimensions = [];
      }
    
      addDimensions(dimension: Dimension) {
        this.dimensions.push(dimension);
      }
}